### DISCUSSION TITLE
How can I apply coincidence counting with logical operations(and, or, not, etc.) like qucoa?

### DISCUSSION THREAD

**Discussion by RemXiel** [PicoQuant/snAPI#20] [Q&A]
I am using the demo of the coincidence counting method in snAPI. The 'manipulator.coincidence' method has a 'chans' parameter that puts the input channel. However, unlike the Qucoa program, we cannot apply logical operations (and, or, not, etc.), so I can't exclude channels other than those I put in.
How can I apply the logical operations to select channels while using snAPI module?

ex) In the coincidence count with a total number of channels of 4, I want to measure [1,3] channels only coincidence count. so it exclude the [1,2,3], [1,3,4], and [1,2,3,4] counts. So I want to measure [1, ¬2, 3, ¬4], (1 & ¬2 & 3 & ¬4).

**Comment by tpoint75** [comment]
Hi, you can't do that with the coincidence manipulator at once. But you remove the counts from Channel 1 and 3 if there is a count on 3 or 4 with the (inverted) herald manipulator and then you can build a coincidence of the left events on Channel 1 and 3:

https://picoquant.github.io/snAPI/snAPI.Main.html#snAPI_Main_Manipulators_herald

This removes the events on channel 1 and 2:
herald(3, [1,2], delayTime=0, gateTime=1000, inverted=True, keepChannels=False)
herald(4, [1,2], delayTime=0, gateTime=1000, inverted=True, keepChannels=False)

and now you can build a coincidence:
coincidence([1,2], windowTime=1000, mode=CoincidenceMode.CountAll, time=CoincidenceTime.Last, keepChannels=False)

..or you do it with without generating extra data in the stream:

This removes the events on channel 1 and 2:
idxNot3  = herald(3, [1,2], delayTime=0, gateTime=1000, inverted=True, keepChannels=True)
idxNot34  = herald(4, idxnot3, delayTime=0, gateTime=1000, inverted=True, keepChannels=True)

and now you can build a coincidence:
idx12Not34 = coincidence(idxNot34, windowTime=1000, mode=CoincidenceMode.CountAll, time=CoincidenceTime.Last, keepChannels=True)

What are your results?
